packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -0,0 +1,161 @@
+      const controllerDirs = this.options.boot.controllerDirs || 'controllers';
+
+      // If custom extensions aren't passed, set default
+      const controllerExts =
+        this.options.boot.controllerExts || 'controller.js';
+
+      // Controllers are Classes so this method can be used to discover
+      // and boot Controllers
+      this.bootClassArtifacts(controllerDirs, controllerExts, this.controller);
+    }
+
+    /**
+     * If an artifact is
+     *
+     * @param dirs List of paths to search for Artifacts. (may be relative to
+     * rootDir / basePath OR absolute paths)
+     * @param exts List of extensions files discovered should match
+     * @param prefixOrFunc String prefix if using a `bind.toClass` or a function
+     * for custom binding logic
+     * @param nested Should nested directories in dirs be searched for Artifacts
+     * @param basePath Override rootDir (for prefixing to relative paths)
+     */
+    bootClassArtifacts(
+      dirs: StringOrArr,
+      exts: StringOrArr,
+      prefixOrFunc: string | Function,
+      nested?: boolean,
+      basePath?: string,
+    ) {
+      basePath = basePath || this.options.boot.rootDir;
+      const files = discoverArtifactFiles(dirs, exts, nested, basePath);
+      if (typeof prefixOrFunc === 'string') {
+        this.bindClassArtifacts(files, prefixOrFunc);
+      } else {
+        this.bindClassArtifactsUsingFunction(files, prefixOrFunc);
+      }
+    }
+
+    /**
+     * Find classes in a list of files and bind them as a Class (to Application
+     * Context) by generating a key using a prefix. Key is: prefix.className
+     *
+     * @param files List of files (absolute paths) to look for Classes in
+     * @param prefix Prefix to generate the binding key (prefix.className)
+     */
+    bindClassArtifacts(files: string[], prefix: string) {
+      files.forEach(file => {
+        try {
+          const ctrl = require(file);
+          const classes: string[] = Object.keys(ctrl);
+
+          classes.forEach((cls: string) =>
+            // prettier-ignore
+            this.bind(`${prefix}.${cls}`).toClass(ctrl[cls]).tag(prefix),
+          );
+        } catch (err) {
+          debug(
+            `bindClassArtifacts: Skipping file: ${file} because error: ${err}`,
+          );
+        }
+      });
+    }
+
+    /**
+     * Find classes in a list of files (absolute paths) and call the given
+     * function with the Classes in the files given.
+     * @param files List of files (absolute paths) to look for Classes in
+     * @param fn Function to pass the Class in to so it can handle "booting" it
+     */
+    bindClassArtifactsUsingFunction(files: string[], fn: Function) {
+      files.forEach(file => {
+        try {
+          const ctrl = require(file);
+          const classes: string[] = Object.keys(ctrl);
+
+          classes.forEach((cls: string) => fn.call(this, ctrl[cls]));
+        } catch (err) {
+          debug(
```

Comment:
The error should be propagated to the caller, not silently hidden to a debug log.

Otherwise the application will start as if everything was good, and then report cryptic errors that would be difficult to troubleshoot, since the root cause - an artifact not loaded/configured - is not part of of those errors and difficult to find (most people run with `debug` disabled on production).